src/components/List.jsx

```diff
@@ -61,23 +62,43 @@ export default function List() {
             </button>
           </div>
         ) : (
-          <ul>
-            {docs.map((item, index) => {
-              return (
-                <li key={index}>
-                  <input
-                    aria-label="checkbox for purchased item"
-                    id={item.data().id}
-                    type="checkbox"
-                    onChange={(e) => handleCheckBox(e, item)}
-                    checked={wasPurchasedWithin24Hours(item)}
-                    disabled={wasPurchasedWithin24Hours(item)}
-                  />
-                  {item.data().item}
-                </li>
-              );
-            })}
-          </ul>
+          <div>
+            <input
+              type="text"
+              placeholder="Search..."
+              value={searchInputValue}
+              onChange={(e) => {
+                setSearchInputValue(e.target.value);
+              }}
+            />
+
+            <button onClick={() => setSearchInputValue(() => '')}>Reset</button>
+
+            <ul>
+              {docs
+                .filter((item) => {
+                  return item
+                    .data()
+                    .item.toLowerCase()
+                    .includes(searchInputValue.toLowerCase());
+                })
+                .map((item, index) => {
+                  return (
+                    <li key={index}>
```

Comment:
This could be a good place to hypothetically refactor with a checkbox component

Reply:
Good call. Thank you for bringing this up. I did gave it a valiant effort of refactoring but was unable to split the component logic. Looking forward to getting the teams feedback on processes on how to achieve this.